storage/src/context.rs

```diff
@@ -52,32 +54,30 @@ impl ContextApi for TezedgeContext {
 
     fn checkout(&self, context_hash: &ContextHash) -> Result<(), ContextError> {
         let mut merkle = self.merkle.write().expect("lock poisoning");
```

Comment:
@michal-stanek 
I also changed order:
```
let mut merkle = self.merkle.write().expect("lock poisoning");
let context_hash_arr: EntryHash = context_hash.as_slice().try_into()?;
```
to 
```
let context_hash_arr: EntryHash = context_hash.as_slice().try_into()?;
let mut merkle = self.merkle.write().expect("lock poisoning");
```

there is no need to do locking, if hash is bad,
but I think this is just a very very very very corner case :)